### PR TITLE
Update to Go 1.20

### PR DESCRIPTION
**Pull request by ItalyPaleAle**
Also updates the pipelines to stop using deprecated Actions.

Note: this also changes the pipelines to read the version of Go to install from the go.mod file. This is useful because otherwise `/ok-to-test` would always install the version of Go defined in the pipeline.

### CODE REVIEW

**Comment by berndverst** [comment]
Earliest date to consider merging this:
Feb 9

Need this page to list Go 1.20, then also verify the weekly GitHub Actions deployment has occurred including that change.
https://github.com/actions/runner-images/blob/main/images/linux/Ubuntu2204-Readme.md#go

**Comment by ItalyPaleAle** [comment]
Looks like Actions does support Go 1.20 now (it's [available](https://raw.githubusercontent.com/actions/go-versions/main/versions-manifest.json), just not cached so the setup step is a bit slower)

However, what I think we should wait for is this: https://github.com/Homebrew/homebrew-core/pull/122082 As without this, a lot of people won't be able to build Dapr locally :)

File: pkg/sentry/certs/certs.go [hunk 1]
```diff
@@ -1,6 +1,7 @@
 package certs
 
 import (
+	"crypto"
```
Comment (mukundansundar) [inline pkg/sentry/certs/certs.go:4]: @ItalyPaleAle can this file change alone be moved to a separate PR? 
And only have at most tests and 1.20 change in this PR?

**Review by mukundansundar — DISMISSED** [review]
lgtm.
Left one small comment.

File: pkg/sentry/certs/certs.go (hunk 1 above)
Comment (ItalyPaleAle) [reply pkg/sentry/certs/certs.go]: Sure, done.

**Comment by ItalyPaleAle** [comment]
@mukundansundar made the change you requested. Sorry for force-pushing but otherwise reverting a single file was a quite complex.

There's no code change now

**Review by berndverst — APPROVED** [review]

**Review by mukundansundar — APPROVED** [review]
